src/Controls/tests/TestCases.HostApp/Issues/Issue25371.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using Microsoft.Maui;
+using Microsoft.Maui.Controls;
+
+namespace Maui.Controls.Sample.Issues
+{
+	[Issue(IssueTracker.Github, 25371, "OnNavigatedTo not called when navigating back to a specific page", PlatformAffected.iOS | PlatformAffected.macOS)]
+	public class Issue25371 : NavigationPage
+	{
+		public Issue25371() : base(new MainPage())
+		{
+		}
+	}
+	public class MainPage : ContentPage
+	{
+		Label label;
+		public MainPage()
+		{
+			var stackLayout = new StackLayout();
+		    label = new Label()
+			{
+				Text = "Welcome to Main page",
+				AutomationId ="MainPageLabel"
+			};
+
+			var button = new Button() { Text = "MoveToNextPage", AutomationId="MoveToNextPage" };
+			button.Clicked += Button_Clicked;
+			stackLayout.Children.Add(label);
+			stackLayout.Children.Add(button);
+			Content = stackLayout;
+			
+		}
+
+		protected override void OnNavigatedTo(NavigatedToEventArgs args)
+		{
+			label.Text = "OnNavigationTo method is called";
+			base.OnNavigatedTo(args);
+		}
+
+		private void Button_Clicked(object sender, EventArgs e)
+		{
+			Navigation.PushAsync(new SecondPage());
+		}
+	}
+
+	public class SecondPage : ContentPage
+	{
+		public SecondPage()
+		{
+			var label = new Label()
+			{
+				AutomationID = "SecondPageLabel",
```

Comment:
```
/Users/builder/azdo/_work/1/s/src/Controls/tests/TestCases.HostApp/Issues/Issue25371.cs(52,5): error CS0117: 'Label' does not contain a definition for 'AutomationID' [/Users/builder/azdo/_work/1/s/src/Controls/tests/TestCases.HostApp/Controls.TestCases.HostApp.csproj::TargetFramework=net9.0-tizen]
/Users/builder/azdo/_work/1/s/src/Controls/tests/TestCases.HostApp/Issues/Issue25371.cs(52,5): error CS0117: 'Label' does not contain a definition for 'AutomationID' [/Users/builder/azdo/_work/1/s/src/Controls/tests/TestCases.HostApp/Controls.TestCases.HostApp.csproj::TargetFramework=net9.0-maccatalyst]
/Users/builder/azdo/_work/1/s/src/Controls/tests/TestCases.HostApp/Issues/Issue25371.cs(52,5): error CS0117: 'Label' does not contain a definition for 'AutomationID' [/Users/builder/azdo/_work/1/s/src/Controls/tests/TestCases.HostApp/Controls.TestCases.HostApp.csproj::TargetFramework=net9.0-ios]
/Users/builder/azdo/_work/1/s/src/Controls/tests/TestCases.HostApp/Issues/Issue25371.cs(52,5): error CS0117: 'Label' does not contain a definition for 'AutomationID' [/Users/builder/azdo/_work/1/s/src/Controls/tests/TestCases.HostApp/Controls.TestCases.HostApp.csproj::TargetFramework=net9.0-android]
    0 Warning(s)
    4 Error(s)
```

Must be **AutomationId**

Reply:
@jsuarezruiz I have changed as AutomationID to AutomationId.